File: baseDatos/Deserializador.java
package baseDatos;

import gestionApp.Empresa;
import gestionApp.Evento.Evento;
import gestionApp.Negocio;
import gestionApp.Venta;
import gestionApp.personas.*;
import gestionApp.servicios.Promocion;
import gestionApp.servicios.Servicio;

import java.io.*;
import java.util.ArrayList;

public class Deserializador {

    public static <E> void deserializador(ArrayList<E> lista, String clase) {
        FileInputStream fileIn;
        try {
            String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + clase + ".txt";
            System.out.println(path);
            File archivo = new File(path);
            archivo.createNewFile();
            fileIn = new FileInputStream(path);
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ArrayList<E> listado = (ArrayList<E>) in.readObject();
            
            for (E owo : listado) {
                lista.add(owo);
            }
            in.close();
            fileIn.close();

        } catch (FileNotFoundException e) {
            e.printStackTrace();

        } catch (IOException e) {
            System.out.println("Esta vacio");

        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }
    }

    public static void serializacion() {
        deserializador(Cliente.getAllClientes(), "Cliente");
        deserializador(Empleado.getAllEmpleados(), "Empleado");
        deserializador(Empresa.getAllEmpresas(), "Empresa");
        deserializador(Negocio.getNegocios(), "Negocio");
        deserializador(Venta.getAllVenta(), "Venta");
        deserializador(Promocion.getAllPromocion(), "Promocion");
        deserializador(Servicio.getAllServicios(), "Servicio");
        deserializador(Evento.getAllEventos(), "Evento");
    }
}

File: baseDatos/Deserializador.py
import pickle
import pathlib
import os

from gestionApp.Empresa import Empresa
from gestionApp.Negocio import Negocio
from gestionApp.Venta import Venta
from gestionApp.eventos.Evento import Evento
from gestionApp.personas.Empleado import Empleado
from gestionApp.personas.Cliente import Cliente
from gestionApp.servicios.Promocion import Promocion
from gestionApp.servicios.Servicio import Servicio


class Deserializador():

    def deserializar():
        datos = {
            "empleado": lambda x: Empleado.setAllEmpleados(x),
            "cliente": lambda x: Cliente.setAllClientes(x),
            "empresa": lambda x: Empresa.setAllEmpresas(x),
            "negocio": lambda x: Negocio.setAllNegocios(x),
            "venta": lambda x: Venta.setAllVentas(x),
            "servicio": lambda x: Servicio.setAllServicios(x),
            "promocion": lambda x: Promocion.setAllPromociones(x),
            "evento": lambda x: Evento.setAllEventos(x)
        }

        for archivo, set in datos.items():
            picklefile = open(os.path.join(pathlib.Path(__file__).parent.absolute(), f"temp\\{archivo}"), "rb")
            dato = pickle.load(picklefile)
            set(dato)
            picklefile.close()
